<!DOCTYPE html>

<html>

<head>

<meta charset="utf-8" />
<meta name="generator" content="pandoc" />
<meta http-equiv="X-UA-Compatible" content="IE=EDGE" />

<meta name="viewport" content="width=device-width, initial-scale=1" />

<meta name="author" content="Christopher Bolen" />

<meta name="date" content="2025-09-02" />

<title>Predicting COO for RNASeq data</title>

<script>// Pandoc 2.9 adds attributes on both header and div. We remove the former (to
// be compatible with the behavior of Pandoc < 2.8).
document.addEventListener('DOMContentLoaded', function(e) {
  var hs = document.querySelectorAll("div.section[class*='level'] > :first-child");
  var i, h, a;
  for (i = 0; i < hs.length; i++) {
    h = hs[i];
    if (!/^h[1-6]$/i.test(h.tagName)) continue;  // it should be a header h1-h6
    a = h.attributes;
    while (a.length > 0) h.removeAttribute(a[0].name);
  }
});
</script>

<style type="text/css">
code{white-space: pre-wrap;}
span.smallcaps{font-variant: small-caps;}
span.underline{text-decoration: underline;}
div.column{display: inline-block; vertical-align: top; width: 50%;}
div.hanging-indent{margin-left: 1.5em; text-indent: -1.5em;}
ul.task-list{list-style: none;}
</style>



<style type="text/css">
code {
white-space: pre;
}
.sourceCode {
overflow: visible;
}
</style>
<style type="text/css" data-origin="pandoc">
pre > code.sourceCode { white-space: pre; position: relative; }
pre > code.sourceCode > span { line-height: 1.25; }
pre > code.sourceCode > span:empty { height: 1.2em; }
.sourceCode { overflow: visible; }
code.sourceCode > span { color: inherit; text-decoration: inherit; }
div.sourceCode { margin: 1em 0; }
pre.sourceCode { margin: 0; }
@media screen {
div.sourceCode { overflow: auto; }
}
@media print {
pre > code.sourceCode { white-space: pre-wrap; }
pre > code.sourceCode > span { display: inline-block; text-indent: -5em; padding-left: 5em; }
}
pre.numberSource code
{ counter-reset: source-line 0; }
pre.numberSource code > span
{ position: relative; left: -4em; counter-increment: source-line; }
pre.numberSource code > span > a:first-child::before
{ content: counter(source-line);
position: relative; left: -1em; text-align: right; vertical-align: baseline;
border: none; display: inline-block;
-webkit-touch-callout: none; -webkit-user-select: none;
-khtml-user-select: none; -moz-user-select: none;
-ms-user-select: none; user-select: none;
padding: 0 4px; width: 4em;
color: #aaaaaa;
}
pre.numberSource { margin-left: 3em; border-left: 1px solid #aaaaaa; padding-left: 4px; }
div.sourceCode
{ }
@media screen {
pre > code.sourceCode > span > a:first-child::before { text-decoration: underline; }
}
code span.al { color: #ff0000; font-weight: bold; } 
code span.an { color: #60a0b0; font-weight: bold; font-style: italic; } 
code span.at { color: #7d9029; } 
code span.bn { color: #40a070; } 
code span.bu { color: #008000; } 
code span.cf { color: #007020; font-weight: bold; } 
code span.ch { color: #4070a0; } 
code span.cn { color: #880000; } 
code span.co { color: #60a0b0; font-style: italic; } 
code span.cv { color: #60a0b0; font-weight: bold; font-style: italic; } 
code span.do { color: #ba2121; font-style: italic; } 
code span.dt { color: #902000; } 
code span.dv { color: #40a070; } 
code span.er { color: #ff0000; font-weight: bold; } 
code span.ex { } 
code span.fl { color: #40a070; } 
code span.fu { color: #06287e; } 
code span.im { color: #008000; font-weight: bold; } 
code span.in { color: #60a0b0; font-weight: bold; font-style: italic; } 
code span.kw { color: #007020; font-weight: bold; } 
code span.op { color: #666666; } 
code span.ot { color: #007020; } 
code span.pp { color: #bc7a00; } 
code span.sc { color: #4070a0; } 
code span.ss { color: #bb6688; } 
code span.st { color: #4070a0; } 
code span.va { color: #19177c; } 
code span.vs { color: #4070a0; } 
code span.wa { color: #60a0b0; font-weight: bold; font-style: italic; } 
</style>
<script>
// apply pandoc div.sourceCode style to pre.sourceCode instead
(function() {
  var sheets = document.styleSheets;
  for (var i = 0; i < sheets.length; i++) {
    if (sheets[i].ownerNode.dataset["origin"] !== "pandoc") continue;
    try { var rules = sheets[i].cssRules; } catch (e) { continue; }
    var j = 0;
    while (j < rules.length) {
      var rule = rules[j];
      // check if there is a div.sourceCode rule
      if (rule.type !== rule.STYLE_RULE || rule.selectorText !== "div.sourceCode") {
        j++;
        continue;
      }
      var style = rule.style.cssText;
      // check if color or background-color is set
      if (rule.style.color === '' && rule.style.backgroundColor === '') {
        j++;
        continue;
      }
      // replace div.sourceCode by a pre.sourceCode rule
      sheets[i].deleteRule(j);
      sheets[i].insertRule('pre.sourceCode{' + style + '}', j);
    }
  }
})();
</script>




<style type="text/css">body {
background-color: #fff;
margin: 1em auto;
max-width: 700px;
overflow: visible;
padding-left: 2em;
padding-right: 2em;
font-family: "Open Sans", "Helvetica Neue", Helvetica, Arial, sans-serif;
font-size: 14px;
line-height: 1.35;
}
#TOC {
clear: both;
margin: 0 0 10px 10px;
padding: 4px;
width: 400px;
border: 1px solid #CCCCCC;
border-radius: 5px;
background-color: #f6f6f6;
font-size: 13px;
line-height: 1.3;
}
#TOC .toctitle {
font-weight: bold;
font-size: 15px;
margin-left: 5px;
}
#TOC ul {
padding-left: 40px;
margin-left: -1.5em;
margin-top: 5px;
margin-bottom: 5px;
}
#TOC ul ul {
margin-left: -2em;
}
#TOC li {
line-height: 16px;
}
table {
margin: 1em auto;
border-width: 1px;
border-color: #DDDDDD;
border-style: outset;
border-collapse: collapse;
}
table th {
border-width: 2px;
padding: 5px;
border-style: inset;
}
table td {
border-width: 1px;
border-style: inset;
line-height: 18px;
padding: 5px 5px;
}
table, table th, table td {
border-left-style: none;
border-right-style: none;
}
table thead, table tr.even {
background-color: #f7f7f7;
}
p {
margin: 0.5em 0;
}
blockquote {
background-color: #f6f6f6;
padding: 0.25em 0.75em;
}
hr {
border-style: solid;
border: none;
border-top: 1px solid #777;
margin: 28px 0;
}
dl {
margin-left: 0;
}
dl dd {
margin-bottom: 13px;
margin-left: 13px;
}
dl dt {
font-weight: bold;
}
ul {
margin-top: 0;
}
ul li {
list-style: circle outside;
}
ul ul {
margin-bottom: 0;
}
pre, code {
background-color: #f7f7f7;
border-radius: 3px;
color: #333;
white-space: pre-wrap; 
}
pre {
border-radius: 3px;
margin: 5px 0px 10px 0px;
padding: 10px;
}
pre:not([class]) {
background-color: #f7f7f7;
}
code {
font-family: Consolas, Monaco, 'Courier New', monospace;
font-size: 85%;
}
p > code, li > code {
padding: 2px 0px;
}
div.figure {
text-align: center;
}
img {
background-color: #FFFFFF;
padding: 2px;
border: 1px solid #DDDDDD;
border-radius: 3px;
border: 1px solid #CCCCCC;
margin: 0 5px;
}
h1 {
margin-top: 0;
font-size: 35px;
line-height: 40px;
}
h2 {
border-bottom: 4px solid #f7f7f7;
padding-top: 10px;
padding-bottom: 2px;
font-size: 145%;
}
h3 {
border-bottom: 2px solid #f7f7f7;
padding-top: 10px;
font-size: 120%;
}
h4 {
border-bottom: 1px solid #f7f7f7;
margin-left: 8px;
font-size: 105%;
}
h5, h6 {
border-bottom: 1px solid #ccc;
font-size: 105%;
}
a {
color: #0033dd;
text-decoration: none;
}
a:hover {
color: #6666ff; }
a:visited {
color: #800080; }
a:visited:hover {
color: #BB00BB; }
a[href^="http:"] {
text-decoration: underline; }
a[href^="https:"] {
text-decoration: underline; }

code > span.kw { color: #555; font-weight: bold; } 
code > span.dt { color: #902000; } 
code > span.dv { color: #40a070; } 
code > span.bn { color: #d14; } 
code > span.fl { color: #d14; } 
code > span.ch { color: #d14; } 
code > span.st { color: #d14; } 
code > span.co { color: #888888; font-style: italic; } 
code > span.ot { color: #007020; } 
code > span.al { color: #ff0000; font-weight: bold; } 
code > span.fu { color: #900; font-weight: bold; } 
code > span.er { color: #a61717; background-color: #e3d2d2; } 
</style>




</head>

<body>




<h1 class="title toc-ignore">Predicting COO for RNASeq data</h1>
<h4 class="author">Christopher Bolen</h4>
<h4 class="date">2025-09-02</h4>



<div id="introduction" class="section level1">
<h1>Introduction</h1>
<p>This vignette details a concise example of running the COO classifier
on a new RNASeq dataset. This example was run on the Rescomp3 HPC
cluster at Genentech, and uses raw data that was output by the Genentech
sequencing pipeline.</p>
</div>
<div id="loading-the-data" class="section level1">
<h1>Loading the data</h1>
<p>This COO classifier requires that you supply the raw count matrix, as
supplied Genentech’s RNASeq processing pipeline. The prediction model
was fit using “stranded count” data, and for best performance we
recommend you use the same dataset. Theoretically this program may work
with unstranded counts, or with data normalized in a different way, but
we can’t guarantee consistent results.</p>
<p>Before we can classify the samples, we have to convert our data to a
DESeqDataSet. For SummarizedExperiments, this is very simple:</p>
<div class="sourceCode" id="cb1"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb1-1"><a href="#cb1-1" tabindex="-1"></a><span class="fu">library</span>(DESeq2)</span>
<span id="cb1-2"><a href="#cb1-2" tabindex="-1"></a>cds <span class="ot">=</span> <span class="fu">DESeqDataSet</span>(strandedCtSummary,<span class="at">design=</span><span class="sc">~</span><span class="dv">1</span>)</span></code></pre></div>
<p>Alternatively, if the data is formatted as a matrix, you could use
the following command:</p>
<div class="sourceCode" id="cb2"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb2-1"><a href="#cb2-1" tabindex="-1"></a><span class="do">## counts is a matrix, with rows of features and columns of samples.</span></span>
<span id="cb2-2"><a href="#cb2-2" tabindex="-1"></a>cds <span class="ot">=</span> <span class="fu">DESeqDataSetFromMatrix</span>(counts,</span>
<span id="cb2-3"><a href="#cb2-3" tabindex="-1"></a>  <span class="at">colData=</span><span class="fu">data.frame</span>(<span class="at">ID=</span><span class="fu">colnames</span>(counts)),</span>
<span id="cb2-4"><a href="#cb2-4" tabindex="-1"></a>  <span class="at">rowData=</span><span class="fu">data.frame</span>(<span class="at">ID=</span><span class="fu">rownames</span>(counts)),</span>
<span id="cb2-5"><a href="#cb2-5" tabindex="-1"></a>  <span class="at">design=</span><span class="sc">~</span><span class="dv">1</span>)</span></code></pre></div>
<p>Note that in order for the classifier to work, it is necessary for
the row names of our CDS to be in the correct format. The classifier
expects the row names to be “Gene IDs”, e.g. “GeneID:9294”, and all
genes used in the model must be present in the dataset. For the standard
model, there are 21 genes, with the following IDs:</p>
<pre><code>##  [1] &quot;`GeneID:10538`&quot;  &quot;`GeneID:10672`&quot;  &quot;`GeneID:11040`&quot;  &quot;`GeneID:152137`&quot;
##  [5] &quot;`GeneID:23139`&quot;  &quot;`GeneID:23495`&quot;  &quot;`GeneID:26278`&quot;  &quot;`GeneID:3055`&quot;  
##  [9] &quot;`GeneID:327657`&quot; &quot;`GeneID:3662`&quot;   &quot;`GeneID:3707`&quot;   &quot;`GeneID:4311`&quot;  
## [13] &quot;`GeneID:4603`&quot;   &quot;`GeneID:51700`&quot;  &quot;`GeneID:54836`&quot;  &quot;`GeneID:55534`&quot; 
## [17] &quot;`GeneID:64764`&quot;  &quot;`GeneID:6597`&quot;   &quot;`GeneID:79754`&quot;  &quot;`GeneID:8994`&quot;  
## [21] &quot;`GeneID:9294`&quot;</code></pre>
</div>
<div id="applying-the-coo-classifier" class="section level1">
<h1>Applying the COO classifier</h1>
<p>The COO classifier can be applied to the cds object using the single
command, coo_rnaseq(). An example is below:</p>
<div class="sourceCode" id="cb4"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb4-1"><a href="#cb4-1" tabindex="-1"></a>pred <span class="ot">=</span> <span class="fu">coo_rnaseq</span>(cds)</span></code></pre></div>
<p>The returned object consists of three columns: - Sample IDs, drawn
from the column names of the cds object. - LPS - the Linear Predictor
Score, used to split samples into GCB and ABC, and - COO - the Cell of
Origin classification; either GCB, ABC, or Unclassified</p>
</div>
<div id="individual-coo-classifier-functions" class="section level1">
<h1>Individual COO classifier functions</h1>
<p>The classifier actually consists of two steps, which can be run
individually. An example is below:</p>
<div id="step-1-normalize" class="section level3">
<h3>Step 1: Normalize</h3>
<div class="sourceCode" id="cb5"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb5-1"><a href="#cb5-1" tabindex="-1"></a>norm <span class="ot">=</span> <span class="fu">coo_normalize</span>(cds)</span></code></pre></div>
</div>
<div id="step-2-apply-the-classifier." class="section level3">
<h3>Step 2: Apply the classifier.</h3>
<p>By default, this uses the built-in “coo_GOYA_21gene” classifier,
developed from the GOYA dataset. However, you have the option of
supplying your own version of the classifier (for example, if not all
genes are available, or if our gene model changes).</p>
<div class="sourceCode" id="cb6"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb6-1"><a href="#cb6-1" tabindex="-1"></a>pred <span class="ot">=</span> <span class="fu">coo_predict</span>(norm_main)</span></code></pre></div>
</div>
</div>
<div id="dzsig-calling" class="section level1">
<h1>DZsig calling</h1>
<p>The DZsig classifier can be applied to the same <code>cds</code>
object:</p>
<div class="sourceCode" id="cb7"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb7-1"><a href="#cb7-1" tabindex="-1"></a>dzsig_pred <span class="ot">=</span> <span class="fu">dzsig_rnaseq</span>(cds)</span></code></pre></div>
<p>Thresholds for the DZsig classification are as published for the
NanoString DLBCL90 assay (Ennishi et al., <em>J Clin Oncol</em> 2019;
Data Supplement).</p>
<p>It is also possible obtain both DZsig and COO calls and generate a
combined “refined COO” classification:</p>
<div class="sourceCode" id="cb8"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb8-1"><a href="#cb8-1" tabindex="-1"></a>refined_pred <span class="ot">=</span> <span class="fu">refined_coo_rnaseq</span>(cds)</span></code></pre></div>
<p>The returned object of this last function has six columns:</p>
<ul>
<li>Sample IDs, drawn from the column names of the cds object.</li>
<li>LPS_COO - the linear predictor score for the COO classification</li>
<li>COO - the Cell of Origin classification; either GCB, ABC, or
Unclassified</li>
<li>LPS_DZsig - the linear predictor score for the DZsig
classification</li>
<li>DZsig - the Dark Zone signature classification; either DZsig-POS,
DZsig-IND, or DZsig-NEG</li>
<li>refined_COO - the refined Cell of Origin classification, which is
assigned hierarchically:
<ul>
<li>If ABC, then ABC</li>
<li>If DZsig-POS, DZsig-POS</li>
<li>Else, COO (GCB or Unclassified)</li>
</ul></li>
</ul>
<p>For more details, please refer to the documentation in the
package.</p>
</div>



<!-- code folding -->


<!-- dynamically load mathjax for compatibility with self-contained -->
<script>
  (function () {
    var script = document.createElement("script");
    script.type = "text/javascript";
    script.src  = "https://mathjax.rstudio.com/latest/MathJax.js?config=TeX-AMS-MML_HTMLorMML";
    document.getElementsByTagName("head")[0].appendChild(script);
  })();
</script>

</body>
</html>
