<!DOCTYPE html>
<!-- Generated by pkgdown: do not edit by hand --><html lang="en">
<head>
<meta http-equiv="Content-Type" content="text/html; charset=UTF-8">
<meta charset="utf-8">
<meta http-equiv="X-UA-Compatible" content="IE=edge">
<meta name="viewport" content="width=device-width, initial-scale=1, shrink-to-fit=no">
<meta name="description" content="This package contains code for applying a Cell of Origin classifier
	to RNASeq data. The process consists of three steps: 1) normalizing, 2) calculating 
	the LPS score, and 3) splitting samples into GCB/ABC based on fixed cut-points.">
<title>An RNASeq-based Cell of Origin classifier • gneSeqCOO</title>
<script src="deps/jquery-3.6.0/jquery-3.6.0.min.js"></script><meta name="viewport" content="width=device-width, initial-scale=1, shrink-to-fit=no">
<link href="deps/bootstrap-5.3.1/bootstrap.min.css" rel="stylesheet">
<script src="deps/bootstrap-5.3.1/bootstrap.bundle.min.js"></script><!-- Font Awesome icons --><link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/5.12.1/css/all.min.css" integrity="sha256-mmgLkCYLUQbXn0B1SRqzHar6dCnv9oZFPEC1g1cwlkk=" crossorigin="anonymous">
<link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/5.12.1/css/v4-shims.min.css" integrity="sha256-wZjR52fzng1pJHwx4aV2AO3yyTOXrcDW7jBpJtTwVxw=" crossorigin="anonymous">
<!-- bootstrap-toc --><script src="https://cdn.jsdelivr.net/gh/afeld/bootstrap-toc@v1.0.1/dist/bootstrap-toc.min.js" integrity="sha256-4veVQbu7//Lk5TSmc7YV48MxtMy98e26cf5MrgZYnwo=" crossorigin="anonymous"></script><!-- headroom.js --><script src="https://cdnjs.cloudflare.com/ajax/libs/headroom/0.11.0/headroom.min.js" integrity="sha256-AsUX4SJE1+yuDu5+mAVzJbuYNPHj/WroHuZ8Ir/CkE0=" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/headroom/0.11.0/jQuery.headroom.min.js" integrity="sha256-ZX/yNShbjqsohH1k95liqY9Gd8uOiE1S4vZc+9KQ1K4=" crossorigin="anonymous"></script><!-- clipboard.js --><script src="https://cdnjs.cloudflare.com/ajax/libs/clipboard.js/2.0.6/clipboard.min.js" integrity="sha256-inc5kl9MA1hkeYUt+EC3BhlIgyp/2jDIyBLS6k3UxPI=" crossorigin="anonymous"></script><!-- search --><script src="https://cdnjs.cloudflare.com/ajax/libs/fuse.js/6.4.6/fuse.js" integrity="sha512-zv6Ywkjyktsohkbp9bb45V6tEMoWhzFzXis+LrMehmJZZSys19Yxf1dopHx7WzIKxr5tK2dVcYmaCk2uqdjF4A==" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/autocomplete.js/0.38.0/autocomplete.jquery.min.js" integrity="sha512-GU9ayf+66Xx2TmpxqJpliWbT5PiGYxpaG8rfnBEk1LL8l1KGkRShhngwdXK1UgqhAzWpZHSiYPc09/NwDQIGyg==" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/mark.js/8.11.1/mark.min.js" integrity="sha512-5CYOlHXGh6QpOFA/TeTylKLWfB3ftPsde7AnmhuitiTX4K5SqCLBeKro6sPS8ilsz1Q4NRx3v8Ko2IBiszzdww==" crossorigin="anonymous"></script><!-- pkgdown --><script src="pkgdown.js"></script><meta property="og:title" content="An RNASeq-based Cell of Origin classifier">
<meta property="og:description" content="This package contains code for applying a Cell of Origin classifier
	to RNASeq data. The process consists of three steps: 1) normalizing, 2) calculating 
	the LPS score, and 3) splitting samples into GCB/ABC based on fixed cut-points.">
<!-- mathjax --><script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.5/MathJax.js" integrity="sha256-nvJJv9wWKEm88qvoQl9ekL2J+k/RWIsaSScxxlsrv8k=" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.5/config/TeX-AMS-MML_HTMLorMML.js" integrity="sha256-84DKXVJXs0/F8OTMzX4UR909+jtl4G7SPypPavF+GfA=" crossorigin="anonymous"></script><!--[if lt IE 9]>
<script src="https://oss.maxcdn.com/html5shiv/3.7.3/html5shiv.min.js"></script>
<script src="https://oss.maxcdn.com/respond/1.4.2/respond.min.js"></script>
<![endif]-->
</head>
<body>
    <a href="#main" class="visually-hidden-focusable">Skip to contents</a>
    

    <nav class="navbar fixed-top navbar-light navbar-expand-lg bg-light"><div class="container">
    
    <a class="navbar-brand me-2" href="index.html">gneSeqCOO</a>

    <small class="nav-text text-muted me-auto" data-bs-toggle="tooltip" data-bs-placement="bottom" title="">1.1.0</small>

    
    <button class="navbar-toggler" type="button" data-bs-toggle="collapse" data-bs-target="#navbar" aria-controls="navbar" aria-expanded="false" aria-label="Toggle navigation">
      <span class="navbar-toggler-icon"></span>
    </button>

    <div id="navbar" class="collapse navbar-collapse ms-3">
      <ul class="navbar-nav me-auto">
<li class="nav-item">
  <a class="nav-link" href="articles/gneSeqCOO.html">Get started</a>
</li>
<li class="nav-item">
  <a class="nav-link" href="reference/index.html">Reference</a>
</li>
      </ul>
<form class="form-inline my-2 my-lg-0" role="search">
        <input type="search" class="form-control me-sm-2" aria-label="Toggle navigation" name="search-input" data-search-index="search.json" id="search-input" placeholder="Search for" autocomplete="off">
</form>

      <ul class="navbar-nav"></ul>
</div>

    
  </div>
</nav><div class="container template-home">
<div class="row">
  <main id="main" class="col-md-9"><div class="section level1">
<div class="page-header"><h1 id="gneseqcoo">gneSeqCOO<a class="anchor" aria-label="anchor" href="#gneseqcoo"></a>
</h1></div>
<p>This package contains code for applying a Cell of Origin classifier to RNASeq data. The process consists of three steps:</p>
<div class="sourceCode" id="cb1"><pre class="sourceCode R"><code class="sourceCode r"><span id="cb1-1"><a href="#cb1-1" tabindex="-1"></a><span class="dv">1</span><span class="er">)</span> normalizing</span>
<span id="cb1-2"><a href="#cb1-2" tabindex="-1"></a><span class="dv">2</span><span class="er">)</span> calculating the LPS score</span>
<span id="cb1-3"><a href="#cb1-3" tabindex="-1"></a><span class="dv">3</span><span class="er">)</span> splitting samples into GCB<span class="sc">/</span>ABC based on fixed cut<span class="sc">-</span>points.</span></code></pre></div>
<p>A parallel process exists for performing Dark Zone signature (DZsig) classification.</p>
</div>
<div class="section level1">
<h1 id="install">Install<a class="anchor" aria-label="anchor" href="#install"></a>
</h1>
<pre><code><span><span class="fu">remotes</span><span class="fu">::</span><span class="fu"><a href="https://remotes.r-lib.org/reference/install_github.html" class="external-link">install_github</a></span><span class="op">(</span><span class="st">"Genentech/gneSeqCOO"</span><span class="op">)</span></span></code></pre>
</div>
<div class="section level1">
<h1 id="basic-usage">Basic usage<a class="anchor" aria-label="anchor" href="#basic-usage"></a>
</h1>
<p>The input for the gneSeqCOO method is a raw count matrix, formatted as a DESeqDataSet. For a standard count matrix, this can be generated via the code below:</p>
<pre><code><span><span class="co">## counts is a matrix, with rows of features and columns of samples.</span></span>
<span><span class="va">cds</span> <span class="op">=</span> <span class="fu">DESeqDataSetFromMatrix</span><span class="op">(</span><span class="va">counts</span>,</span>
<span>  colData<span class="op">=</span><span class="fu"><a href="https://rdrr.io/r/base/data.frame.html" class="external-link">data.frame</a></span><span class="op">(</span>ID<span class="op">=</span><span class="fu"><a href="https://rdrr.io/r/base/colnames.html" class="external-link">colnames</a></span><span class="op">(</span><span class="va">counts</span><span class="op">)</span><span class="op">)</span>,</span>
<span>  rowData<span class="op">=</span><span class="fu"><a href="https://rdrr.io/r/base/data.frame.html" class="external-link">data.frame</a></span><span class="op">(</span>ID<span class="op">=</span><span class="fu"><a href="https://rdrr.io/r/base/colnames.html" class="external-link">rownames</a></span><span class="op">(</span><span class="va">counts</span><span class="op">)</span><span class="op">)</span>,</span>
<span>  design<span class="op">=</span><span class="op">~</span><span class="fl">1</span><span class="op">)</span></span></code></pre>
<p>Note that in order for the classifier to work, it is necessary for the row names of our CDS to be in the correct format. The classifier currently accepts Refseq “Gene IDs” (formatted as “GeneID:9294”) or Ensembl IDs (ENSG….). Genes in the dataset should not be filtered before running the algorithm, as the complete genome is used to normalize individual samples.</p>
<p>The COO classifier can be applied to the cds object using the single command, coo_rnaseq(). An example is below:</p>
<pre><code><span><span class="va">coo_pred</span> <span class="op">=</span> <span class="fu"><a href="reference/coo_rnaseq.html">coo_rnaseq</a></span><span class="op">(</span><span class="va">cds</span><span class="op">)</span></span></code></pre>
<p>The returned object consists of three columns:</p>
<ul>
<li>Sample IDs, drawn from the column names of the cds object.</li>
<li>LPS - the Linear Predictor Score, used to split samples into GCB and ABC, and</li>
<li>COO - the Cell of Origin classification; either GCB, ABC, or Unclassified</li>
</ul>
<p>It is also possible obtain both DZsig and COO calls and generate a combined “refined COO” classification:</p>
<pre><code><span><span class="va">refined_pred</span> <span class="op">=</span> <span class="fu"><a href="reference/refined_coo_rnaseq.html">refined_coo_rnaseq</a></span><span class="op">(</span><span class="va">cds</span><span class="op">)</span></span></code></pre>
<p>The returned object of this last function has six columns:</p>
<ul>
<li>Sample IDs, drawn from the column names of the cds object.</li>
<li>LPS_COO - the linear predictor score for the COO classification</li>
<li>COO - the Cell of Origin classification; either GCB, ABC, or Unclassified</li>
<li>LPS_DZsig - the linear predictor score for the DZsig classification</li>
<li>DZsig - the Dark Zone signature classification; either DZsig-POS, DZsig-IND, or DZsig-NEG</li>
<li>refined_COO - the refined Cell of Origin classification, which is assigned hierarchically:
<ul>
<li>If ABC, then ABC</li>
<li>If DZsig-POS, DZsig-POS</li>
<li>Else, COO (GCB or Unclassified)</li>
</ul>
</li>
</ul>
<p>For more details, please refer to the documentation in the package.</p>
</div>

  </main><aside class="col-md-3"><div class="license">
<h2 data-toc-skip>License</h2>
<ul class="list-unstyled">
<li><a href="LICENSE.html">Full license</a></li>
<li><small><a href="https://opensource.org/licenses/mit-license.php" class="external-link">MIT</a> + file <a href="LICENSE-text.html">LICENSE</a></small></li>
</ul>
</div>


<div class="citation">
<h2 data-toc-skip>Citation</h2>
<ul class="list-unstyled">
<li><a href="authors.html#citation">Citing gneSeqCOO</a></li>
</ul>
</div>

<div class="developers">
<h2 data-toc-skip>Developers</h2>
<ul class="list-unstyled">
<li>Chris Bolen <br><small class="roles"> Author, maintainer </small>  </li>
</ul>
</div>



  </aside>
</div>


    <footer><div class="pkgdown-footer-left">
  <p></p>
<p>Developed by Chris Bolen.</p>
</div>

<div class="pkgdown-footer-right">
  <p></p>
<p>Site built with <a href="https://pkgdown.r-lib.org/" class="external-link">pkgdown</a> 2.0.7.</p>
</div>

    </footer>
</div>

  

  

  </body>
</html>
