<!DOCTYPE html>
<!-- Generated by pkgdown: do not edit by hand --><html lang="en">
<head>
<meta http-equiv="Content-Type" content="text/html; charset=UTF-8">
<meta charset="utf-8">
<meta http-equiv="X-UA-Compatible" content="IE=edge">
<meta name="viewport" content="width=device-width, initial-scale=1, shrink-to-fit=no">
<meta name="description" content="gneSeqCOO">
<title>Predicting COO for RNASeq data • gneSeqCOO</title>
<script src="../deps/jquery-3.6.0/jquery-3.6.0.min.js"></script><meta name="viewport" content="width=device-width, initial-scale=1, shrink-to-fit=no">
<link href="../deps/bootstrap-5.2.2/bootstrap.min.css" rel="stylesheet">
<script src="../deps/bootstrap-5.2.2/bootstrap.bundle.min.js"></script><!-- Font Awesome icons --><link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/5.12.1/css/all.min.css" integrity="sha256-mmgLkCYLUQbXn0B1SRqzHar6dCnv9oZFPEC1g1cwlkk=" crossorigin="anonymous">
<link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/5.12.1/css/v4-shims.min.css" integrity="sha256-wZjR52fzng1pJHwx4aV2AO3yyTOXrcDW7jBpJtTwVxw=" crossorigin="anonymous">
<!-- bootstrap-toc --><script src="https://cdn.jsdelivr.net/gh/afeld/bootstrap-toc@v1.0.1/dist/bootstrap-toc.min.js" integrity="sha256-4veVQbu7//Lk5TSmc7YV48MxtMy98e26cf5MrgZYnwo=" crossorigin="anonymous"></script><!-- headroom.js --><script src="https://cdnjs.cloudflare.com/ajax/libs/headroom/0.11.0/headroom.min.js" integrity="sha256-AsUX4SJE1+yuDu5+mAVzJbuYNPHj/WroHuZ8Ir/CkE0=" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/headroom/0.11.0/jQuery.headroom.min.js" integrity="sha256-ZX/yNShbjqsohH1k95liqY9Gd8uOiE1S4vZc+9KQ1K4=" crossorigin="anonymous"></script><!-- clipboard.js --><script src="https://cdnjs.cloudflare.com/ajax/libs/clipboard.js/2.0.6/clipboard.min.js" integrity="sha256-inc5kl9MA1hkeYUt+EC3BhlIgyp/2jDIyBLS6k3UxPI=" crossorigin="anonymous"></script><!-- search --><script src="https://cdnjs.cloudflare.com/ajax/libs/fuse.js/6.4.6/fuse.js" integrity="sha512-zv6Ywkjyktsohkbp9bb45V6tEMoWhzFzXis+LrMehmJZZSys19Yxf1dopHx7WzIKxr5tK2dVcYmaCk2uqdjF4A==" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/autocomplete.js/0.38.0/autocomplete.jquery.min.js" integrity="sha512-GU9ayf+66Xx2TmpxqJpliWbT5PiGYxpaG8rfnBEk1LL8l1KGkRShhngwdXK1UgqhAzWpZHSiYPc09/NwDQIGyg==" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/mark.js/8.11.1/mark.min.js" integrity="sha512-5CYOlHXGh6QpOFA/TeTylKLWfB3ftPsde7AnmhuitiTX4K5SqCLBeKro6sPS8ilsz1Q4NRx3v8Ko2IBiszzdww==" crossorigin="anonymous"></script><!-- pkgdown --><script src="../pkgdown.js"></script><meta property="og:title" content="Predicting COO for RNASeq data">
<meta property="og:description" content="gneSeqCOO">
<!-- mathjax --><script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.5/MathJax.js" integrity="sha256-nvJJv9wWKEm88qvoQl9ekL2J+k/RWIsaSScxxlsrv8k=" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.5/config/TeX-AMS-MML_HTMLorMML.js" integrity="sha256-84DKXVJXs0/F8OTMzX4UR909+jtl4G7SPypPavF+GfA=" crossorigin="anonymous"></script><!--[if lt IE 9]>
<script src="https://oss.maxcdn.com/html5shiv/3.7.3/html5shiv.min.js"></script>
<script src="https://oss.maxcdn.com/respond/1.4.2/respond.min.js"></script>
<![endif]-->
</head>
<body>
    <a href="#main" class="visually-hidden-focusable">Skip to contents</a>
    

    <nav class="navbar fixed-top navbar-light navbar-expand-lg bg-light"><div class="container">
    
    <a class="navbar-brand me-2" href="../index.html">gneSeqCOO</a>

    <small class="nav-text text-muted me-auto" data-bs-toggle="tooltip" data-bs-placement="bottom" title="">1.0.0</small>

    
    <button class="navbar-toggler" type="button" data-bs-toggle="collapse" data-bs-target="#navbar" aria-controls="navbar" aria-expanded="false" aria-label="Toggle navigation">
      <span class="navbar-toggler-icon"></span>
    </button>

    <div id="navbar" class="collapse navbar-collapse ms-3">
      <ul class="navbar-nav me-auto">
<li class="active nav-item">
  <a class="nav-link" href="../articles/gneSeqCOO.html">Get started</a>
</li>
<li class="nav-item">
  <a class="nav-link" href="../reference/index.html">Reference</a>
</li>
      </ul>
<form class="form-inline my-2 my-lg-0" role="search">
        <input type="search" class="form-control me-sm-2" aria-label="Toggle navigation" name="search-input" data-search-index="../search.json" id="search-input" placeholder="Search for" autocomplete="off">
</form>

      <ul class="navbar-nav"></ul>
</div>

    
  </div>
</nav><div class="container template-article">




<div class="row">
  <main id="main" class="col-md-9"><div class="page-header">
      <img src="" class="logo" alt=""><h1>Predicting COO for RNASeq data</h1>
                        <h4 data-toc-skip class="author">Christopher
Bolen</h4>
            
            <h4 data-toc-skip class="date">2024-08-15</h4>
      
      
      <div class="d-none name"><code>gneSeqCOO.Rmd</code></div>
    </div>

    
    
<div class="section level2">
<h2 id="introduction">Introduction<a class="anchor" aria-label="anchor" href="#introduction"></a>
</h2>
<p>This vignette details a concise example of running the COO classifier
on a new RNASeq dataset. This example was run on the Rescomp3 HPC
cluster at Genentech, and uses raw data that was output by the Genentech
sequencing pipeline.</p>
</div>
<div class="section level2">
<h2 id="loading-the-data">Loading the data<a class="anchor" aria-label="anchor" href="#loading-the-data"></a>
</h2>
<p>This COO classifier requires that you supply the raw count matrix, as
supplied Genentech’s RNASeq processing pipeline. The prediction model
was fit using “stranded count” data, and for best performance we
recommend you use the same dataset. Theoretically this program may work
with unstranded counts, or with data normalized in a different way, but
we can’t guarantee consistent results.</p>
<p>Before we can classify the samples, we have to convert our data to a
DESeqDataSet. For SummarizedExperiments, this is very simple:</p>
<div class="sourceCode" id="cb1"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="kw"><a href="https://rdrr.io/r/base/library.html" class="external-link">library</a></span><span class="op">(</span><span class="va"><a href="https://github.com/thelovelab/DESeq2" class="external-link">DESeq2</a></span><span class="op">)</span></span>
<span><span class="va">cds</span> <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/pkg/DESeq2/man/DESeqDataSet.html" class="external-link">DESeqDataSet</a></span><span class="op">(</span><span class="va">strandedCtSummary</span>,design<span class="op">=</span><span class="op">~</span><span class="fl">1</span><span class="op">)</span></span></code></pre></div>
<p>Alternatively, if the data is formatted as a matrix, you could use
the following command:</p>
<div class="sourceCode" id="cb2"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="co">## counts is a matrix, with rows of features and columns of samples.</span></span>
<span><span class="va">cds</span> <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/pkg/DESeq2/man/DESeqDataSet.html" class="external-link">DESeqDataSetFromMatrix</a></span><span class="op">(</span><span class="va">counts</span>,</span>
<span>  colData<span class="op">=</span><span class="fu"><a href="https://rdrr.io/r/base/data.frame.html" class="external-link">data.frame</a></span><span class="op">(</span>ID<span class="op">=</span><span class="fu"><a href="https://rdrr.io/r/base/colnames.html" class="external-link">colnames</a></span><span class="op">(</span><span class="va">counts</span><span class="op">)</span><span class="op">)</span>,</span>
<span>  rowData<span class="op">=</span><span class="fu"><a href="https://rdrr.io/r/base/data.frame.html" class="external-link">data.frame</a></span><span class="op">(</span>ID<span class="op">=</span><span class="fu"><a href="https://rdrr.io/r/base/colnames.html" class="external-link">rownames</a></span><span class="op">(</span><span class="va">counts</span><span class="op">)</span><span class="op">)</span>,</span>
<span>  design<span class="op">=</span><span class="op">~</span><span class="fl">1</span><span class="op">)</span></span></code></pre></div>
<p>Note that in order for the classifier to work, it is necessary for
the row names of our CDS to be in the correct format. The classifier
expects the row names to be “Gene IDs”, e.g. “GeneID:9294”, and all
genes used in the model must be present in the dataset. For the standard
model, there are 21 genes, with the following IDs:</p>
<pre><code><span><span class="co">##  [1] "`GeneID:10538`"  "`GeneID:10672`"  "`GeneID:11040`"  "`GeneID:152137`"</span></span>
<span><span class="co">##  [5] "`GeneID:23139`"  "`GeneID:23495`"  "`GeneID:26278`"  "`GeneID:3055`"  </span></span>
<span><span class="co">##  [9] "`GeneID:327657`" "`GeneID:3662`"   "`GeneID:3707`"   "`GeneID:4311`"  </span></span>
<span><span class="co">## [13] "`GeneID:4603`"   "`GeneID:51700`"  "`GeneID:54836`"  "`GeneID:55534`" </span></span>
<span><span class="co">## [17] "`GeneID:64764`"  "`GeneID:6597`"   "`GeneID:79754`"  "`GeneID:8994`"  </span></span>
<span><span class="co">## [21] "`GeneID:9294`"</span></span></code></pre>
</div>
<div class="section level2">
<h2 id="applying-the-coo-classifier">Applying the COO classifier<a class="anchor" aria-label="anchor" href="#applying-the-coo-classifier"></a>
</h2>
<p>The COO classifier can be applied to the cds object using the single
command, coo_rnaseq(). An example is below:</p>
<div class="sourceCode" id="cb4"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">pred</span> <span class="op">=</span> <span class="fu"><a href="../reference/coo_rnaseq.html">coo_rnaseq</a></span><span class="op">(</span><span class="va">cds</span><span class="op">)</span></span></code></pre></div>
<p>The returned object consists of three columns: - Sample IDs, drawn
from the column names of the cds object. - LPS - the Linear Predictor
Score, used to split samples into GCB and ABC, and - COO - the Cell of
Origin classification; either GCB, ABC, or Unclassified</p>
</div>
<div class="section level2">
<h2 id="individual-coo-classifier-functions">Individual COO classifier functions<a class="anchor" aria-label="anchor" href="#individual-coo-classifier-functions"></a>
</h2>
<p>The classifier actually consists of two steps, which can be run
individually. An example is below:</p>
<div class="section level4">
<h4 id="step-1-normalize">Step 1: Normalize<a class="anchor" aria-label="anchor" href="#step-1-normalize"></a>
</h4>
<div class="sourceCode" id="cb5"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">norm</span> <span class="op">=</span> <span class="fu"><a href="../reference/coo_normalize.html">coo_normalize</a></span><span class="op">(</span><span class="va">cds</span><span class="op">)</span></span></code></pre></div>
</div>
<div class="section level4">
<h4 id="step-2-apply-the-classifier-">Step 2: Apply the classifier.<a class="anchor" aria-label="anchor" href="#step-2-apply-the-classifier-"></a>
</h4>
<p>By default, this uses the built-in “coo_GOYA_21gene” classifier,
developed from the GOYA dataset. However, you have the option of
supplying your own version of the classifier (for example, if not all
genes are available, or if our gene model changes).</p>
<div class="sourceCode" id="cb6"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">pred</span> <span class="op">=</span> <span class="fu"><a href="../reference/coo_predict.html">coo_predict</a></span><span class="op">(</span><span class="va">norm_main</span><span class="op">)</span></span></code></pre></div>
</div>
</div>
  </main><aside class="col-md-3"><nav id="toc"><h2>On this page</h2>
    </nav></aside>
</div>



    <footer><div class="pkgdown-footer-left">
  <p></p>
<p>Developed by Chris Bolen.</p>
</div>

<div class="pkgdown-footer-right">
  <p></p>
<p>Site built with <a href="https://pkgdown.r-lib.org/" class="external-link">pkgdown</a> 2.0.7.</p>
</div>

    </footer>
</div>

  

  

  </body>
</html>
